### PR TITLE
Forbid custom namespaces in various resources

### PR DESCRIPTION
**Pull request by codingllama**
Forbid the use of custom namespaces in various resources, akin to commit ae99259. Only "" or "default" (as in apidefaults.Namespace) are allowed.

Namespaces are a vestigial feature from olden Teleport times. It's largely deprecated/unused, but this PR gives us some extra confidence to eventually do a wider cleanup in the future.

(Note: this is not about k8s namespaces.)

#49509

### CODE REVIEW

**Comment by codingllama** [comment]
@fspmarshall let me know if you think I missed something.

File: api/types/presence.go [hunk 1]
```diff
@@ -70,6 +70,9 @@ func (s *KeepAlive) CheckAndSetDefaults() error {
 	if s.Namespace == "" {
 		s.Namespace = defaults.Namespace
 	}
+	if s.Namespace != defaults.Namespace {
+		return trace.BadParameter("namespace %q invalid, custom namespaces are deprecated", s.Namespace)
+	}
```
Comment (fspmarshall) [inline api/types/presence.go:75]: nit: in this and the other error messages like it, we probably want to explicitly tell the user how to fix the problem.  Ex:

```suggestion
	if s.Namespace != defaults.Namespace {
		return trace.BadParameter("namespace %q invalid, custom namespaces are deprecated, the namespace field should be omitted or set to %q", s.Namespace, defaults.Namespace)
	}
```

**Review by fspmarshall — APPROVED** [review]

File: api/types/presence.go (hunk 1 above)
Comment (codingllama) [reply api/types/presence.go]: Done using a validation helper. I'm not even sure users can set the field (in most resources the json tag is set to "-"), but code definitely can (and some resources seem to use protojson which bypasses the gogojson tag).

**Comment by codingllama** [comment]
PTAL @fspmarshall.

**Review by rosstimothy — APPROVED** [review]

**Comment by codingllama** [comment]
Thanks for the quick reviews, folks.

**Pull request merged**